bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java

```diff
@@ -323,5 +362,37 @@ public String getURL() {
             }
         }
 
+        public String getId() {
+            return parentId + "-cta-" + StringUtils.substring(DigestUtils.sha1Hex(path), 0, 10);
```

Comment:
see comment above for AbstractListItemImpl

Reply:
The ID generation differs from a regular component because it requires the parentId.
There is an abstract class for ListItems but that one uses <parentId>-item-<hash> and this one requires <parentId>-cta-<hash>